instance/context.go

```diff
@@ -301,18 +305,22 @@ func DumpAssets(jobSpec models.JobSpec, scheduledAt time.Time, engine models.Tem
 			return nil, err
 		}
 		assetsToDump = compiledAssetResponse.Assets.ToJobSpec().ToMap()
+		skipCompile = compiledAssetResponse.SkipCompile
 	}
 
 	// compile again if needed
-	templates, err := engine.CompileFiles(assetsToDump, map[string]interface{}{
-		ConfigKeyDstart:        jobSpec.Task.Window.GetStart(scheduledAt).Format(models.InstanceScheduledAtTimeLayout),
-		ConfigKeyDend:          jobSpec.Task.Window.GetEnd(scheduledAt).Format(models.InstanceScheduledAtTimeLayout),
-		ConfigKeyExecutionTime: scheduledAt.Format(models.InstanceScheduledAtTimeLayout),
-		ConfigKeyDestination:   jobDestination,
-	})
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to compile templates")
+	if !skipCompile {
```

Comment:
`DumpAssets` is used during dryrun of the job in validation step. I am afraid if during that stage it is not compiled properly, validation will fail. We will need to skip validation of these specs somehow